File: Simulators/simulators/services/rmm.js
/*
 * Copyright (c) 2015-2017 Intel Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the 'License');
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an 'AS IS' BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const Service = require('./service').Service;
const ServiceRoot = require('../api/rmm/serviceRoot').RmmServiceRoot;
const Rmc = require('../api/rmm/managers/rmc').Rmc;
const UUID = require('uuid');

class Rmm extends Service {
    constructor(port = '7812', uuid = UUID.v1(), discoveryProtocols = ["dhcp"]) {
        super(uuid, discoveryProtocols, port, 'rmm', '/redfish/v1');

        this.serviceRoot = new ServiceRoot(this.mock, this.serviceRoot.uuid);

        this.eventTrigger = this.serviceRoot.EventTrigger;
        this.rack = this.serviceRoot.AddRack();
        this.drawer = this.AddDrawer();
        this.drawer.BindToRack(this.rack);
        this.thermalZone = this.rack.AddThermalZone();
        this.thermalZone.AddFan();

        this.powerZone = this.rack.AddPowerZone();
        this.powerZone.AddPsu();

        this.thermal = this.rack.Thermal;
        var thermalFan = this.thermal.AddThermalFan();
        var redundancy = this.thermal.AddRedundancy();
        var thermalTemp = this.thermal.AddThermalTemperature();

        redundancy.AddToRedundancySet(thermalFan);
        thermalTemp.AddToRelatedItems(this.rack);
        thermalFan.AddToRelatedItems(this.rack);
        thermalFan.AddToRedundancySet(redundancy);

        var rmc = this.serviceRoot.AddRmc();
        rmc.AddManagerForChassis(this.rack);
        rmc.AddEthernetInterface().AddVlan();

        var mbpc = this.serviceRoot.AddMbpc();

        var mbp = this.rack.AddMbp();
        mbpc.AddMbpToManager(mbp);


    }

    get EventTrigger() {
        return this.serviceRoot.EventTrigger;
    }

    get Rack() {
        return this.rack;
    }

    get Drawers() {
        return this.serviceRoot.Drawers;
    }

    get ThermalZone() {
        return this.thermalZone;
    }

    get PowerZone() {
        return this.powerZone;
    }

    AddDrawer() {
        return this.serviceRoot.AddDrawer();
    }

    RemoveDrawerWithId(id) {
        return this.serviceRoot.RemoveDrawerWithId(id);
    }

    get ServiceRoot() {
        return this.serviceRoot;
    }

    get Managers() {
        return this.serviceRoot.Managers;
    }
}

exports.Rmm = Rmm;

File: PSME/common/agent-framework/src/module/constants/rmm.cpp
/*!
 * @copyright
 * Copyright (c) 2017-2019 Intel Corporation
 *
 * @copyright
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * @copyright
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * @copyright
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 *
 * @file rmm.cpp
 * @brief Contains RMM constants used all over the model
 * */

#include "agent-framework/module/constants/rmm.hpp"

namespace agent_framework {
namespace model {
namespace literals {

constexpr const char Fan::STATUS[];
constexpr const char Fan::CURRENT_SPEED[];
constexpr const char Fan::FRU_INFO[];
constexpr const char Fan::OEM[];
constexpr const char Fan::FAN[];
constexpr const char Fan::PHYSICAL_CONTEXT[];
constexpr const char Fan::CURRENT_SPEED_UNITS[];

constexpr const char ThermalZone::ZONE[];
constexpr const char ThermalZone::STATUS[];
constexpr const char ThermalZone::VOLUMETRIC_AIRFLOW_CFM[];
constexpr const char ThermalZone::DESIRED_SPEED_PWM[];
constexpr const char ThermalZone::COLLECTIONS[];
constexpr const char ThermalZone::OEM[];

constexpr const char Psu::PSU[];
constexpr const char Psu::STATUS[];
constexpr const char Psu::FRU_INFO[];
constexpr const char Psu::OEM[];
constexpr const char Psu::POWER_SUPPLY_TYPE[];
constexpr const char Psu::LINE_INPUT_VOLTAGE_TYPE[];
constexpr const char Psu::LINE_INPUT_VOLTAGE_VOLTS[];
constexpr const char Psu::FIRMWARE_VERSION[];
constexpr const char Psu::LAST_POWER_OUTPUT_WATTS[];
constexpr const char Psu::POWER_CAPACITY_WATTS[];
constexpr const char Psu::INDICATOR_LED[];
constexpr const char Psu::REQUESTED_STATE[];

constexpr const char PowerZone::ZONE[];
constexpr const char PowerZone::STATUS[];
constexpr const char PowerZone::POWER_ALLOCATED[];
constexpr const char PowerZone::POWER_REQUESTED[];
constexpr const char PowerZone::POWER_AVAILABLE[];
constexpr const char PowerZone::POWER_CONSUMED[];
constexpr const char PowerZone::POWER_CAPACITY[];
constexpr const char PowerZone::COLLECTIONS[];
constexpr const char PowerZone::OEM[];

constexpr const char ChassisSensor::SENSOR[];
constexpr const char ChassisSensor::STATUS[];
constexpr const char ChassisSensor::READING[];
constexpr const char ChassisSensor::PHYSICAL_CONTEXT[];
constexpr const char ChassisSensor::READING_UNITS[];
constexpr const char ChassisSensor::SENSOR_NUMBER[];
constexpr const char ChassisSensor::OEM[];

}
}
}
